validator/client/wait_for_activation.go

```diff
@@ -87,47 +88,87 @@ func (v *validator) waitForActivation(ctx context.Context, accountsChangedChan <
 		time.Sleep(time.Second * time.Duration(mathutil.Min(uint64(attempts), 60)))
 		return v.waitForActivation(incrementRetries(ctx), accountsChangedChan)
 	}
-	for {
-		select {
-		case <-accountsChangedChan:
-			// Accounts (keys) changed, restart the process.
-			return v.waitForActivation(ctx, accountsChangedChan)
-		default:
-			res, err := stream.Recv()
-			// If the stream is closed, we stop the loop.
-			if errors.Is(err, io.EOF) {
-				break
-			}
-			// If context is canceled we return from the function.
+
+	remoteKm, ok := v.keyManager.(remote.RemoteKeymanager)
+	if ok {
```

Comment:
The diff here looks like a mess, but I basically moved all existing code into the `else` branch and implemented the `if` branch.